raiden_contracts/contracts/services/OneToN.sol

```diff
@@ -0,0 +1,125 @@
+pragma solidity ^0.5.2;
+
+import "raiden/Utils.sol";
+import "services/UserDeposit.sol";
+import "raiden/lib/ECVerify.sol";
+
+contract OneToN is Utils {
+    string constant public contract_version = "0.5.0";
+
+    UserDeposit deposit_contract;
+
+    // Indicates which sessions have already been settled by storing
+    // keccak256(receiver, sender, expiration_block) => expiration_block.
+    mapping (bytes32 => uint256) public settled_sessions;
+
+    /*
+     *  Events
+     */
+
+    // The session has been settled and can't be claimed again. The receiver is
+    // indexed to allow services to know when claims have been successfully
+    // processed.
+    // When users want to get notified about low balances, they should listen
+    // for UserDeposit.BalanceReduced, instead.
+    // The first three values identify the session, `transferred` is the amount
+    // of tokens that has actually been transferred during the claim.
+    event Claimed(
+        address sender,
+        address indexed receiver,
+        uint256 expiration_block,
+        uint256 transferred
+    );
+
+    /*
+     *  Constructor
+     */
+
+    /// @param _deposit_contract Address of UserDeposit contract
+    constructor(address _deposit_contract)
+        public
+    {
+        deposit_contract = UserDeposit(_deposit_contract);
+    }
+
+    /// @notice Submit an IOU to claim the owed amount.
+    /// If the deposit is smaller than the claim, the remaining deposit is
+    /// claimed. If no tokens are claimed, `claim` may be retried, later.
+    /// @param sender Address from which the amount is transferred
+    /// @param receiver Address to which the amount is transferred
+    /// @param amount Owed amount of tokens
+    /// @param expiration_block Tokens can only be claimed before this time
+    /// @param signature Sender's signature over keccak256(sender, receiver, amount, expiration_block)
+    /// @return Amount of transferred tokens
+    function claim(
+        address sender,
+        address receiver,
+        uint256 amount,
+        uint256 expiration_block,
+        bytes memory signature
+    )
+        public
+        returns (uint)
+    {
+        require(block.number <= expiration_block);
+
+        // validate signature
+        address addressFromSignature = recoverAddressFromSignature(
+            sender,
+            receiver,
+            amount,
+            expiration_block,
+            signature
+        );
+        require(addressFromSignature == sender);
+
+        // must not be claimed before
+        bytes32 _key = keccak256(abi.encodePacked(receiver, sender, expiration_block));
```

Comment:
Why is `_key` prefixed with an underscore?  (Pure curiosity.)

Reply:
Heiko did that when sketching his idea and I thought it might be a keyword and left it that way. But looking it up, it's not one of the reserved keywords: https://solidity.readthedocs.io/en/v0.4.21/miscellaneous.html#reserved-keywords . So the "key" word is not a keyword.